### PR TITLE
Update Target to readonly and apply the in modifier.

### PR DESCRIPTION
**Pull request by teinarss**
According to https://docs.microsoft.com/en-us/dotnet/csharp/write-safe-efficient-code#apply-the-in-modifier-to-readonly-struct-parameters-larger-than-systemintptrsize we should use the `in` modifier to readonly structs that are larger than IntPtr.Size (which is 8 byte), Our `Target` struct is 72 bytes in size.

Also did a benchmark, did not show any big (or at all?) performance improvement, we should see a bigger improvement in a game with many orders running. 
image

### CODE REVIEW

**Comment by pchote** [comment]
Needs a rebase: no merge conflict, but we have some new code that needs to be updated to avoid compile errors.

File: OpenRA.Game/GameRules/WeaponInfo.cs
Comment (pchote) [inline OpenRA.Game/GameRules/WeaponInfo.cs]: Can we have a comment here like `// Lambdas can't use in variables, so capture a copy for later` and rename this to something like `delayedTarget`?

File: OpenRA.Mods.Common/Activities/Attack.cs
Comment (pchote) [inline OpenRA.Mods.Common/Activities/Attack.cs]: Likewise here, maybe `rangeTarget`?

File: OpenRA.Mods.Common/Traits/Air/Aircraft.cs
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Air/Aircraft.cs]: maybe cache `var targetActor = target.Actor` and use it in both places below?

File: OpenRA.Mods.Common/Traits/Armament.cs
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Armament.cs]: `delayedTarget`?

**Review by pchote — DISMISSED** [review]
I'd like to do something about the `target1` names, otherwise LGTM.

**Comment by teinarss** [comment]
Updated

**Comment by pchote** [comment]
Can we please get a second review and merge here before it needs another rebase? If we aren't careful we will otherwise end up with a broken bleed compile due to incompatible changes that don't explicitly cause a merge conflict.

File: OpenRA.Game/GameRules/WeaponInfo.cs
Comment (reaperrr) [inline OpenRA.Game/GameRules/WeaponInfo.cs]: Nit: "be used" instead of "use"?

File: OpenRA.Mods.Common/Activities/Attack.cs
Comment (reaperrr) [inline OpenRA.Mods.Common/Activities/Attack.cs]: Like above.

File: OpenRA.Game/GameRules/WeaponInfo.cs
Comment (reaperrr) [reply OpenRA.Game/GameRules/WeaponInfo.cs]: Nevermind, got the grammar now.

Comment (pchote) [reply OpenRA.Game/GameRules/WeaponInfo.cs]: Maybe change these to 
> // Lambdas can't use "in" variables, so capture a copy for later

to avoid confusion?

Comment (reaperrr) [reply OpenRA.Game/GameRules/WeaponInfo.cs]: That would be better, yes.

**Comment by abcdefg30** [comment]
Rebased (first push) and applied fixups (following push).

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/e096a1cf5a0cff0848efeabd9e37d3d424447688...adfd9c8b796be8d4480efa1d6a1f5a868ab3615b)

**Pull request merged**
